### DISCUSSION TITLE
How to safely create a `VarLenUnicode`?

### DISCUSSION THREAD

**Discussion by Kriskras99** [metno/hdf5-rust#42] [Q&A] [answered]
The `VarLenUnicode` type can only be created using two functions:

- [`new() -> VarLenUnicode`](https://docs.rs/hdf5-metno/latest/hdf5_metno/types/struct.VarLenUnicode.html#method.new) which creates an empty string.
- [`unsafe from_str_unchecked(S) -> VarLenUnicode`](https://docs.rs/hdf5-metno/latest/hdf5_metno/types/struct.VarLenUnicode.html#method.from_str_unchecked) with no documentation on the safety requirements.

I'm currently using [`VarLenAscii::from_ascii`](https://docs.rs/hdf5-metno/latest/hdf5_metno/types/struct.VarLenAscii.html#method.from_ascii) and just hoping that my data never contains unicode.

What's the correct way of doing this?

**Comment by magnusuMET** [comment, marked as answer]
[`FromStr::from`](https://docs.rs/hdf5-metno/0.9.3/hdf5_metno/types/struct.VarLenUnicode.html#impl-FromStr-for-VarLenUnicode) should be what you are looking for

**Reply by Kriskras99** [reply]
Awesome, thanks!
